athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -186,9 +186,7 @@ public ListTablesResponse doListTables(BlockAllocator allocator, ListTablesReque
                 .filter(index -> !index.startsWith("."));
 
         // get all data streams from ES, 1 data stream can contains multiple indices which start with ".ds-xxxxxxxxxx"
-        Stream<String> dataStreams = client.indices().getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT).getDataStreams()
-                .stream()
-                .map(dataStream -> dataStream.getName());
+        Stream<String> dataStreams = getDataStreamNamesFromClient(client);
 
         //combine two different data sources and create tables
         List<TableName> tableNames = Stream.concat(indicesStream, dataStreams)
```

Comment:
If you aren't using it in more than this one line, omit 189 and then just replace this line with `List<TableName> tableNames = Stream.concat(indicesStream, getDataStreamNames(client))`